### DISCUSSION TITLE
J4125 Fanless Soft Router Mini PC 4x2.5G Intel NICs

### DISCUSSION THREAD

**Discussion by taxbeday** [SmoothWAN/SmoothWAN#49] [Ideas and feature requests]
@TalalMash your solution is amazing! Thank you for putting this together. This solves a bunch of problems for me: (1) Speedify on an always on-solution, (2) VPN through Speedify; (3) VPN through Starlink; (4) VPN through LTE; and router. 

I came across SmoothWAN, looking for a solution for a StarLink setup with an LTE modem backup. Since the pandemic I implemented and regularly use Speedify for a specific cannot-fail video streaming use case, so I am very familiar with Speedify. I was looking for a solution for an allways-on router with Speedify and stumbled upon SmoothWAN. 

As a test, I have an Intel Z8350 based mini PC with one integrated Realtek LAN, I plugged in 2 more Realtek-based USB ethernet dongles, DD'd the x86 image on the Z8350, setup Speedify, and it just works. What I need is an always on solution, and this wonky dongled solution will not work long term. 

Based upon the Z8350 success, I ordered this  J4125 Fanless Soft Router Mini PC from Ali for $128 (barebones):

https://aliexpress.com/item/3256803916388691.html

This has 4 intel 2.5G LAN NICs. If this works I will be very happy.

**Comment by TalalMash** [comment]
Hi! Thanks, It should work out of the box, you may have to guess which port is the LAN port. I've included most Intel modules, let me know if there is an issue.

**Reply by taxbeday** [reply]
Yes, I noticed that kmod-igc is included in the build script. I will update once the unit arrives. 

Regarding assignment of WAN1 and WAN2 ports, I assume that at first boot you have to guess which is the LAN port. But, once logged in you can assign ports by MAC and upon shutdown and reboot the port assignments are sticky?

**Reply by TalalMash** [reply]
@taxbeday Yes, the MAC app is redundant for modular hardware, it's not needed, the assignment should be sticky by the kernel.

**Reply by taxbeday** [reply]
> @taxbeday Yes, the MAC app is redundant for modular hardware, it's not needed, the assignment should be sticky by the kernel.

OK great - I will report back when I receive the hardware and install everything.

**Reply by sdacheson78** [reply]
This is what I'm doing.  I run Proxmox on my Qotom 8 NIC box with 6 USB.  Works like a champ

**Comment by taxbeday** [comment]
I received the devkce above in the OP. It is as advertised. I am having a problem with SmoothWan in both the X86 build and via the VMWare build running in proxmox. The issue is that about 50% of the time on boot (bare metal or VM), when OpenWRT is loading it throws a driver not found error for one or more of the 4 Ethernet ports. It appears to do so more on reboot than from cold boot. It appears to be a SmoothWan or OpenWRT issue with this particular box, because while running vanilla linux or proxmox this problem does not occur. 

The issue may be this:

https://forum.openwrt.org/t/2-5g-intel-ethernet-controller-i225-v-not-working-cont/118155

or this:

https://github.com/astlinux-project/astlinux/commit/4ab028f300887d63a67ee02f2a5f384610c461c0

Still troubleshooting...

**Reply by TalalMash** [reply]
Let me know if you've found a fix/workaround, I'll apply it asap.
A screenshot of the error would be helpful.
I couldn't replicate the issue in the VM, were you using passthrough?

**Reply by taxbeday** [reply]
This occurs, same error, in bare metal and proxmox with the same error relating to unable to load drivers for the i225 - but again, it might load 2 of the i225 cards fine, but one of the i225 does not. Still working on it.

**Comment by taxbeday** [comment]
I am trying to figure out how to apply the changes to mod-igc here:

https://git.kernel.org/pub/scm/linux/kernel/git/tnguy/next-queue.git/commit/drivers/net/ethernet/intel/igc?h=dev-queue&id=7c496de538eebd8212dc2a3c9a468386b264d0d4

also here:

https://github.com/astlinux-project/astlinux/commit/4ab028f300887d63a67ee02f2a5f384610c461c0

and build with those changes. It appears that the mod-igc for the kernel of the current openwrt build does not incorporate the changes above. See the various iterations of mod-igc here:

https://www.kernelconfig.io/config_igc?q=&kernelversion=5.4.210&arch=x86

**Reply by TalalMash** [reply]
The patch is incompatible, particularly igc_phy, couldn't build it, I'm surprised that it affects Proxmox too, can you change the virtual network adapter?

**Reply by taxbeday** [reply]
Thank you for your efforts on this. I will try SmoothWAN on Proxmox without passing through the i225 NICs. That should work. I will report back.

It looks like this will be a problem with the i225 NIC until after the newer changes to mod-igc are incorporated.

**Reply by TalalMash** [reply]
No problem, the new OpenWRT snapshots use a recent kernel and the patches worked, will have to port/fix SmoothWAN related changes.

**Comment by taxbeday** [comment]
Confirmed that the box in the OP works fine with proxmox as host with SmoothWan as VM with the i225 NICs in bridge mode.

**Reply by TalalMash** [reply]
Thanks

**Comment by taxbeday** [comment]
This was my first effort with proxmox. I might have setup something wrong, but I was having all kinds of issues (latency and hanging) with Proxmox and SmoothWan with this setup. 

I am very experienced with ESXi so I tried ESXi instead. With ESXi, performance appears to be way better. It might be an i225 driver issue with proxmox, or I might have configured something wrong. ESXi requires you to slipstream the community driver for i225 support for this device, but so far it's working great. 

Proxmox includes i225 support out of the box. 

I'll follow up after more time with the ESXi setup.

**Reply by TalalMash** [reply]
I've also noticed poorer performance with kvm and VirtualBox.
